### ISSUE TITLE
check resnet implementation here and compare difference: https://github.com/KaiyangZhou/Dassl.pytorch/blob/c61a1b570ac6333bd50fb5ae06aea59002fb20bb/dassl/modeling/backbone/__init__.py

### ISSUE THREAD

**Issue by smilesun** [marrlab/DomainLab#264]
https://github.com/KaiyangZhou/Dassl.pytorch/blob/c61a1b570ac6333bd50fb5ae06aea59002fb20bb/dassl/modeling/backbone/__init__.py

we might also want to check if there are some architectural differences of the resnet there compared to our implementation which can result in a performance difference in accuracy:

https://github.com/marrlab/DomainLab/issues/233

**Comment by Car-la-F** [comment]
for now I just looked into resnet50

It seems like Dassl uses an implementation which is oriented on a implementation in Pytorch which can also be found in the webside of pytorch (https://pytorch.org/vision/0.8/_modules/torchvision/models/resnet.html), the code is nearly the same and they seem to download the weights from the url `https://download.pytorch.org/models/resnet50-19c8e357.pth`. However it seems like the code from pytorch is not the one normally used to download a resnet, I couln't find anything about the advantage doing it like this.

When looking into our implementation I find that we use normal the pytorch implementation of resnet5 (which is also referenced in their documentation https://pytorch.org/vision/main/_modules/torchvision/models/resnet.html), it is not the one which is used by Dassl. the weights loaded into our resnet50 model do originate from the url `https://download.pytorch.org/models/resnet50-11ad3fa6.pth` which is different to the one above.

However I don't know what is the reason for Dassl to use a diferent implementation as the one referenced in the documentation. It is likely that the only difference are the loaded weights, which might originate from a different training.

**Comment by smilesun** [comment]
can we try to have this weight to see if the results differ? Because many results really depend on the initilization .

**Comment by Car-la-F** [comment]
@smilesun I wrote some code to get a resnet with the same weights as Dassl and put it into #274
